File: src/q_2020/junior/CyclicShifts.java
package q_2020.junior;

import java.util.Scanner;

public class CyclicShifts {

	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		String str1 = input.nextLine();
		String str2 = input.nextLine();
		String result = "no";
		for(int i = 0; i< str2.length(); i++) {
			str2 = str2.substring(1) + str2.charAt(0);
			if(str1.indexOf(str2) != -1) {
				result = "yes";
			}
		}
		System.out.println(result);
	}

}

File: src/q_2020/junior/cyclicshifts.py
str1 = input('')
str2 = input('')
result = 'no'
for i in range(len(str2)):
    str2 = str2[1:len(str2)] + str2[0]
    if str1.count(str2) > 0:
        result = 'yes'
        break
print(result)
    